design/coreos.md

```diff
@@ -0,0 +1,209 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# First class support for CoreOS images
+
+## Status
+
+implementable
+
+## Summary
+
+This design proposes adding first class support for CoreOS images and ramdisks,
+while retaining all features of the Ironic backend.
+
+Note: *CoreOS* in this document implies both Fedora CoreOS and Red Hat CoreOS.
+the former will be used in Metal3 by default.
+
+## Motivation
+
+Currently deploying CoreOS is possible both through a normal workflow and by
+attaching a CoreOS image as a live ISO and running `coreos-installer` from it.
+
+Compared to the normal Ironic workflow this proposal:
+
+- Gets rid of special ramdisk images, normal CoreOS artefacts are used.
```

Comment:
```suggestion
- Gets rid of special ramdisk images, normal CoreOS artifacts are used.
```

Reply:
Not sure if there's a language/style guide for the docs here, but note that `artefact` is a valid British spelling (British being the flavour many Europeans were taught): https://www.merriam-webster.com/dictionary/artefact

Reply:
Ah that's a new one to me! I'm happy to leave it, but the rest of the doc is in US english (i.e. customization). Sorry for bike shedding, thought it was a typo.

Reply:
Haha no worries! Looks like Dmitry switched to the US spelling anyway. I only mention it because it trips up US folks more than some of the other spelling differences do.

Reply:
I'm chaotic good, my spelling is erratic!